src/main/scala/uk/gov/nationalarchives/dp/FileProcessors.scala

```diff
@@ -0,0 +1,114 @@
+package uk.gov.nationalarchives.dp
+
+import cats.effect.IO
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import fs2.{Chunk, text}
+import fs2.interop.reactivestreams.PublisherOps
+import org.reactivestreams.Publisher
+import uk.gov.nationalarchives.DAS3Client
+import uk.gov.nationalarchives.dp.client.AdminClient
+import uk.gov.nationalarchives.dp.client.FileInfo._
+import cats.implicits._
+import io.circe.{Decoder, HCursor}
+import io.circe.parser.decode
+import java.nio.ByteBuffer
+import scala.jdk.CollectionConverters.ListHasAsScala
+import scala.xml.XML
+
+object FileProcessors {
+  implicit class KeyUtils(s: String) {
+    def getName: String = s.split("/").last
+
+    def withoutSuffix: String = s.getName.split("\\.").head
+  }
+
+  case class S3Objects(objects: List[S3Object])
+  case class S3Object(bucket: String, key: String)
+
+  implicit val decodeObject: Decoder[S3Object] = (c: HCursor) =>
+    for {
+      key <- c.downField("s3").downField("object").downField("key").as[String]
+      bucket <- c.downField("s3").downField("bucket").downField("name").as[String]
+    } yield {
+      S3Object(bucket, key)
+    }
+
+  implicit val decodeS3: Decoder[S3Objects] = (c: HCursor) =>
+    for {
+      objects <- c.downField("Records").as[List[S3Object]]
+    } yield {
+      S3Objects(objects)
+    }
+
+  private[dp] def processSchemas(key: String, xmlData: String): SchemaFileInfo =
+    SchemaFileInfo(key.withoutSuffix, "", key.getName, xmlData)
+
+  private[dp] def processTransforms(key: String, xmlData: String): TransformFileInfo = {
+    val name = key.withoutSuffix
+    val fromNameSpace = XML.loadString(xmlData).getNamespace("tns")
+    val purpose = if (name.contains("view")) "view" else "edit"
+    TransformFileInfo(name, fromNameSpace, "http://www.w3.org/1999/xhtml", purpose, key.getName, xmlData)
+  }
+
+  private[dp] def processIndexDefinitions(key: String, xmlData: String): IndexDefinitionInfo =
+    IndexDefinitionInfo(key.withoutSuffix, xmlData)
+
+  private[dp] def processMetadataTemplates(key: String, xmlData: String): MetadataTemplateInfo =
+    MetadataTemplateInfo(key.withoutSuffix, xmlData)
+
+  private[dp] def processFiles(
+      client: AdminClient[IO],
+      s3Client: DAS3Client[IO],
+      secretName: String,
+      s3Objects: List[S3Object]
+  ): IO[List[Unit]] = {
+    s3Objects
+      .map(event => {
+        val key = event.key
+        for {
+          publisher <- s3Client.download(event.bucket, key)
+          xml <- xmlFromPublisher(publisher)
+          _ <- key.split("/").head match {
```

Comment:
On the line above, just so people in the future know what it is, maybe you could save this (`key.split("/").head`) to a variable name? Or maybe add it as another method to the `KeyUtils` implicit class like the `getName` method?

Reply:
Yeah, good point. I've added in a `documentType` method to the `KeyUtils` class and used that here.